cmd/builder/internal/builder/templates/main.go.tmpl

```diff
@@ -35,14 +35,21 @@ func main() {
 					{{.Name}}.NewFactory(),
 					{{- end}}
 				},
-				ConverterFactories: []confmap.ConverterFactory{
-					expandconverter.NewFactory(),
-				},
 			},
 		},
 		{{- end}}
 	}
 
+    {{ if .Distribution.SupportsConfmapFactories }}
+	if confmap.UseStableExpansionRules.IsEnabled() {
+		set.ConfigProviderSettings.ResolverSettings.DefaultScheme = "env"
+	} else {
+		set.ConfigProviderSettings.ResolverSettings.ConverterFactories = []confmap.ConverterFactory{
+			expandconverter.NewFactory(),
+		}
+	}
+    {{- end}}
```

Comment:
@mx-psi @evan-bradley this solution has the downside that it will only apply to people that use ocb.  Is that ok?

Does it need to live in `otelcol`, like in `NewCommand`, and be allowed to mess with the incoming settings, ignoring any configured expandconverter?

Reply:
If possible, this is the process I think would work best:

1. Implement the feature gate in `NewCommand`/`NewCollector`/etc. with what you have here.
2. Turn the feature gate on by default.
3. Allow configuring the default scheme in an ocb manifest.
4. Remove the feature gate and the expand converter from ocb in one go, since there's no way to configure converters in the ocb manifest right now.

Reply:
If the two behaviors can coexist, I don't think we should remove a configured expand converter.

Reply:
Thinking about this again I propose a alternative:

Since this is a change to how `CollectorSettings` is configured, we handle the transition for only our own usage (ocb) - anyone using `NewCommand` is responsible for updating `CollectorSettings` on their own, following the deprecation of the `expandconverter`.

This approach would mean moving the featuregate into the `main.go` generated by ocb, deprecating `expandconverter` and then as part of that deprecation following a standard feature gate process.

This approach saves us the trouble of guessing how other users of `NewCommand` want to handle the deprecation of the `expandconverter` while at the same time enforcing our own Provider expectations established in the rfc.

Reply:
We would first have to handle [this TODO](https://github.com/open-telemetry/opentelemetry-collector/blob/ed767dc21968b2b2a2ba2ac95f40fe510de9de79/otelcol/command.go#L56) first, where we start requiring incoming `CollectorSettings` to have the providers/converts supplied, making `NewCommand` more dependent on a full set of settings when invoked.

Reply:
@evan-bradley @mx-psi @codeboten I am still working towards this PR but trying to unblock/simplify it by working on https://github.com/open-telemetry/opentelemetry-collector/issues/10290 first.

Reply:
Thinking about this some more, I believe by moving the feature gate to builder and only controlling our own use of `expandconverter` allows this PR and https://github.com/open-telemetry/opentelemetry-collector/pull/10359 to happen in parallel.